src/ripple/app/tx/impl/Transactor.cpp

```diff
@@ -1395,6 +1406,16 @@ Transactor::doHookCallback(
         if (hookObj.getFieldH256(sfHookHash) != callbackHookHash)
             continue;
 
+        // default allows all transaction types
+        uint256 defaultHookEmit = UINT256_BIT[ttHOOK_SET];
+
+        uint256 hookEmit =
+            (hookObj.isFieldPresent(sfHookEmit)
+                 ? hookObj.getFieldH256(sfHookEmit)
```

Comment:
Maybe a helper function for this

Reply:
https://github.com/Xahau/xahaud/pull/392/commits/225e02afed95664d968e37c5b5a4292a9e887e06